tests/v4l2encode.cpp

```diff
@@ -147,23 +148,23 @@ bool feedOneInputFrame(int fd, int index = -1 /* if index is not -1, simple enqu
 
 bool writeOneOutputFrame(uint8_t* data, uint32_t dataSize)
 {
-    static FILE* fp = NULL;
+    std::ifstream m_ofs = NULL;
 
-    if(!fp) {
-        fp = fopen(outputFileName, "w+");
-        if (!fp) {
+    if(!m_ofs) {
```

Comment:
Need to check `if(!m_ofs.is_open())` here.  `!m_ofs` only checks if failbit or badbit is set... afaik, these bits aren't set on a default constructed ofstream.  The only time you should use `!m_ofs` is after attempting to open a file or an operation has been performed on an opened file.

Reply:
gtk!